src/main/java/reposense/parser/CsvParser.java

```diff
@@ -58,7 +58,7 @@ public CsvParser(Path csvFilePath) throws IOException {
             String line;
 
             for (int lineNumber = 2; (line = br.readLine()) != null; lineNumber++) {
-                String[] elements = line.split(ELEMENT_SEPARATOR);
+                String[] elements = line.split(ELEMENT_SEPARATOR_REGEX);
 
```

Comment:
The trim and dealing of this should be placed in CSV parser, directly after this line. You should make sure all the elements have the same context as the user's intended one.